resin/logs.py

```diff
@@ -1,133 +1,188 @@
         """
-        This function allows fetching historical device logs.
-
-        Args:
-            uuid (str): device uuid.
-            callback (function): this callback is called on receiving a message from the channel.
-            error (function): this callback is called on an error event.
-            For more details about callbacks in pubnub subscribe, visit here: https://www.pubnub.com/docs/python/api-reference#history
-
-        Examples:
-            # Define callback and error.
-            >>> def callback(message):
-            ...     print(message)
-            >>> def error(message):
-            ...     print('Error:'+ str(message))
-            Logs.history(uuid=uuid, callback=callback, error=error)
 
-        """
+        self.device.get(uuid)
+        if uuid is not None and uuid not in self.subscriptions:
+            self.subscriptions[uuid] = self.subscription_handler.add(uuid, callback, error, count)
 
-        _print_deprecation_warning()
-        channel = self.get_channel(uuid)
-        self.pubnub.history(channel=channel, callback=callback, error=error)
+        self.subscription_handler.run()
 
-    def unsubscribe(self, uuid):
+    def history(self, uuid, count=None):
         """
-        This function allows unsubscribing to device logs.
+        Get device logs history.
 
         Args:
             uuid (str): device uuid.
+            count (Optional[int]): this callback is called on receiving a message from the channel.
 
         """
 
-        _print_deprecation_warning()
-        if hasattr(self, 'pubnub'):
-            channel = self.get_channel(uuid)
-            self.pubnub.unsubscribe(channel=channel)
+        raw_query = ''
+
+        if count:
+            raw_query = 'count={}'.format(count)
 
-    def get_channel(self, uuid):
+        return self.base_request.request(
+            '/device/v2/{uuid}/logs'.format(uuid=uuid), 'GET', raw_query=raw_query,
+            endpoint=self.settings.get('api_endpoint')
+        )
+
+    def unsubscribe(self, uuid=None):
         """
-        This function returns pubnub channel for a specific device.
+        Unsubscribe from device logs.
+        If there is no device uuid then all subscribed devices will be unsubscribed.
 
         Args:
-            uuid (str): device uuid.
-
-        Returns:
-            str: device channel.
+            uuid (Optional[str]): device uuid.
 
         """
 
-        _print_deprecation_warning()
-        if not hasattr(self, 'logs_channel'):
-            device_info = self.device.get(uuid)
-            if 'logs_channel' in device_info:
-                self.logs_channel = device_info['logs_channel']
-            else:
-                self.logs_channel = uuid
-
-        return 'device-{logs_channel}-logs'.format(logs_channel=self.logs_channel)
+        if uuid and uuid in self.subscriptions:
+            self.subscription_handler.stop(self.subscriptions[uuid])
+            self.subscriptions.pop(uuid, None)
```

Comment:
No need to `pop` if you're not using the result. Use `del self.subscriptions[uuid]` instead

Reply:
IMO, I prefer this way since it won't raise any exception if uuid doesn't exist in the dict but we already have a check for that case so it's fine.
Updated.